src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,44 @@
+package duke;
+
+import java.util.Scanner;
+
+/**
+ * The Duke class represents the chatbot.
+ */
+public class Duke {
+    /** Stores file information and handles the file operations **/
+    private Storage storage;
+    /** Stores the list of tasks **/
+    private TaskList tasks;
+    /** Instance of Ui to handle user interactions **/
+    private Ui ui;
+
+    /**
+     * Instantiates an instance of Duke.
+     *
+     * @param filePath String of the intended file path.
+     */
+    public Duke(String filePath) {
+        this.ui = new Ui();
+        this.storage = new Storage(filePath);
+        this.tasks = new TaskList(storage.load());
+    }
+
+    /**
+     * Runs the program.
+     */
+    public void run() {
+        Ui.start();
+        Scanner scan = new Scanner(System.in);
+        String input = Ui.getInput(scan);
+        while (!input.equals("bye")) {
+            Parser.parseInput(input, this.tasks, this.storage);
+            input = Ui.getInput(scan);
+        }
+        Ui.bye();
```

Comment:
remember to close the scanner.

perhaps you can bundle the scanner as part of the Ui class for better encapsulation.

-> you can instantiate the scanner in Ui. and when you call the bye function, you can close the scanner and print the goodbye message.